### PR TITLE
Convert tooling-api to kotlin dsl

### PR DESCRIPTION
**Pull request by wolfs**

### CODE REVIEW

**Comment by eskatos** [comment]
Feedback build: https://builds.gradle.org/viewQueued.html?itemId=11723733

File: subprojects/tooling-api/tooling-api.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import accessors.*
+import org.gradle.gradlebuild.BuildEnvironment
+import org.gradle.gradlebuild.buildquality.testfiles.TestFileCleanUpExtension
+import org.gradle.gradlebuild.packaging.ShadedJar
+import org.gradle.gradlebuild.test.integrationtests.IntegrationTest
+import org.gradle.gradlebuild.unittestandcompile.ModuleType
+import org.gradle.plugins.ide.eclipse.model.EclipseClasspath
```
Comment (eskatos) [inline subprojects/tooling-api/tooling-api.gradle.kts]: :x: unused import

File: subprojects/tooling-api/tooling-api.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+val integTestTasks: DomainObjectCollection<IntegrationTest> by extra
+
+integTestTasks.all {
+    binaryDistributions.binZipRequired = true
+    libsRepository.required = true
+}
+
+the<TestFileCleanUpExtension>().isErrorWhenNotEmpty = false
```
Comment (eskatos) [inline subprojects/tooling-api/tooling-api.gradle.kts]: :x: should use the generated static accessor:
```kotlin
testFilesCleanup {
    isErrorWhenNotEmpty = false
}
```

File: subprojects/tooling-api/eclipse.gradle [hunk 3]
```diff
@@ -0,0 +1,10 @@
+eclipse {
```
Comment (eskatos) [inline subprojects/tooling-api/eclipse.gradle]: :x: should stay inlined in the `.gradle.kts` script, similar logic exists in `IdePlugin` for inspiration

File: subprojects/tooling-api/tooling-api.gradle.kts (hunk 1 above)
Comment (wolfs) [reply subprojects/tooling-api/tooling-api.gradle.kts]: Fixed.

Why don't I see the unused imports in IDEA?

File: subprojects/tooling-api/tooling-api.gradle.kts (hunk 2 above)
Comment (wolfs) [reply subprojects/tooling-api/tooling-api.gradle.kts]: Fixed.

File: subprojects/tooling-api/eclipse.gradle (hunk 3 above)
Comment (wolfs) [reply subprojects/tooling-api/eclipse.gradle]: I included the script.

**Pull request merged**